test/integration/targets/user/tasks/main.yml

```diff
@@ -429,6 +429,52 @@
           - bsd_account_expiration.stdout == '0'
   when: ansible_os_family == 'FreeBSD'
 
+# Test setting epoch 0 expiration when creating a new account, then removing the expiry
+# https://github.com/ansible/ansible/issues/47114
+- name: Remove ansibulluser
+  user:
+    name: ansibulluser
+    state: absent
+
+- name: Create user account with epoch 0 expiration
+  user:
+    name: ansibulluser
+    state: present
+    expires: 0
```

Comment:
Run this again to test no changes are report. Currently, it always reports a change.

Reply:
Seems like it happens only on FreeBSD though? :/

Reply:
I was able to get it to report changes every time testing on a Linux VM.

Reply:
OK, I was able to reproduce it on Linux as well. Changed the fix as per your suggestion. I left the FreeBSD portion of the code intact as the issue this is fixing is reported against Linux.